web/src/vdom.ts

```diff
@@ -2,7 +2,26 @@ import _ from "lodash";
 
 import * as blueslip from "./blueslip";
 
-export function eq_array(a, b, eq) {
+type Equal<T> = (...args: T[]) => boolean;
```

Comment:
`eq` for `Node` takes 1 argument, `eq` for `eq_array` takes 2 arguments. Let's type them separately to be more precise.